### PR TITLE
Perf: picoev handle response buffer

### PR DESCRIPTION
**Pull request by enghitalo**
Motivated by the benchmark below and some discussions on discord

```v
import benchmark
import picohttpparser
import time

const max_iterations = 99_999_999

fn main() {

	mut buf_out_from_picoev := unsafe { malloc_noscan(max_iterations * 10) }
	mut buf := []u8{cap: max_iterations * 10}

	mut res := picohttpparser.Response{
		// fd: 0
		buf_start:  buf_out_from_picoev
		buf:  buf_out_from_picoev
		date: gen_date().str
	}

	mut b := benchmark.start()

	for i := 0; i < max_iterations; i++ {
		res.write_string("0123456789")
		
	}
unsafe{ free(res.buf) }

	b.measure('write_string')
	for i := 0; i < max_iterations; i++ {
		value := "0123456789"
		unsafe { buf.push_many(value.str, value.len) }
		
	}
unsafe{ free(buf) }
	b.measure('push_many')
}

fn gen_date() string {
	gmt := time.utc()
	mut date := gmt.strftime('---, %d --- %Y %H:%M:%S GMT')
	date = date.replace_once('---', gmt.weekday_str())
	date = date.replace_once('---', gmt.smonth())
	return "date"
}
```
There was no difference in performance with wrk to run 2 picoev projects from example folder
`wrk -H 'Connection: keep-alive' --connections 64 --threads 8 -d 10 http://localhost:8080`

### CODE REVIEW

**Comment by spytheman** [comment]
Why are you freeing in a loop?

**Comment by enghitalo** [comment]
I was testing outside the loop, when I placed the loop, I forgot to change the position of the free.

**Review by Casper64 — CHANGES_REQUESTED** [review]

File: vlib/picohttpparser/response.c.v [hunk 1]
```diff
@@ -8,16 +8,15 @@ pub struct Response {
 pub:
 	fd        int
 	date      &u8 = unsafe { nil }
-	buf_start &u8 = unsafe { nil }
+	buf_start []u8
 pub mut:
-	buf &u8 = unsafe { nil }
+	buf []u8
```
[the remark this answers is not in the document]
Comment (enghitalo) [reply vlib/picohttpparser/response.c.v:13]: The plan, soon, is to not have to do these casts. I didn't have time to finish, so I uploaded the part I had already done.

[the remark this answers is not in the document]
Comment (enghitalo) [reply vlib/picohttpparser/response.c.v:13]: The plan, soon, is to not have to do these casts. I didn't have time to finish, so I uploaded the part I had already done.

**Comment by Casper64** [comment]
My guess is that you didn't sent enough requests to picoev, so the benchmarks gave equal results. When I run your branch with the picoev example in `examples/picoev/pico.v` I can see a clear difference in test speeds between your branch and the current code. About at least a 15% performance decrease.

When I tests I always make sure the cpu usage is 100%, if this isn't the case the loop can take more requests per second and you will not see a difference. Also I assume you compiled with `-prod`?

Below are the average out of 3 runs for each V version each.

Current V:
```
wrk -t 12 -c 400 -d 15s http://localhost:8089
Running 15s test @ http://localhost:8089
  12 threads and 400 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     0.92ms    1.68ms 106.25ms   99.78%
    Req/Sec    38.46k     8.61k  253.59k    85.56%
  6944841 requests in 15.10s, 834.51MB read
Requests/sec: 459952.33
Transfer/sec:     55.27MB
```

Your branch:
```
wrk -t 12 -c 400 -H 'Connection: keep-alive' -d 15s http://localhost:8089
Running 15s test @ http://localhost:8089
  12 threads and 400 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency     1.21ms    2.13ms 119.89ms   99.72%
    Req/Sec    29.53k    12.31k  244.39k    60.95%
  5312063 requests in 15.10s, 638.31MB read
Requests/sec: 351791.27
Transfer/sec:     42.27MB
```

**Pull request closed without merging**